File: app/src/main/cpp/model/DigitalOwner.cpp
//
// Created by kiber_god on 21.09.2023.
//

#include "DigitalOwner.h"

DigitalOwner::DigitalOwner() {
    dayTriggering = DEFAULT_DAY_TRIGGERING;
    monthTriggering = DEFAULT_MONTH_TRIGGERING;
    yearTriggering = DEFAULT_YEAR_TRIGGERING;
    mode = DEFAULT_MODE;
}

DigitalOwner::DigitalOwner(const int dayTriggering, const int monthTriggering,
                           const int yearTriggering, const int mode) {
    this->dayTriggering = dayTriggering;
    this->monthTriggering = monthTriggering;
    this->yearTriggering = yearTriggering;
    this->mode = mode;
}

const int DigitalOwner::getDayTriggering() const {
    return dayTriggering;
}

const int DigitalOwner::getMonthTriggering() const {
    return monthTriggering;
}

const int DigitalOwner::getYearTriggering() const {
    return yearTriggering;
}

const int DigitalOwner::getMode() const {
    return mode;
}

File: app/src/main/java/com/kibergod/passwordstorage/model/DigitalOwner.java
package com.kibergod.passwordstorage.model;


/*
 *  Клас "Цифрового власника"
 */
public class DigitalOwner {

    public static final int PASSIVE_MODE = 0;          // Режим бездії
    public static final int HIDE_MODE = 1;             // Режим приховування даних
    public static final int PROTECTED_MODE = 2;        // Режим захисту входу
    public static final int DATA_DELETION_MODE = 3;    // Режим знищення даних

    public static final int DEFAULT_DAY_TRIGGERING = 0;
    public static final int DEFAULT_MONTH_TRIGGERING = 0;
    public static final int DEFAULT_YEAR_TRIGGERING = 0;

    public static boolean retrieveRecords = false;

    private int dayTriggering;
    private int monthTriggering;
    private int yearTriggering;

    private int mode;

    public DigitalOwner(int dayTriggering, int monthTriggering, int yearTriggering, int mode) {
        this.dayTriggering = dayTriggering;
        this.monthTriggering = monthTriggering;
        this.yearTriggering = yearTriggering;
        this.mode = mode;
    };

    public int getDayTriggering() { return dayTriggering; }
    public int getMonthTriggering() { return monthTriggering; }
    public int getYearTriggering() { return yearTriggering; }

    public int getMode() { return mode; }

    public void setMode(int mode) {
        this.mode = mode;
    }

    public void setDate(int day, int month, int year) {
        this.dayTriggering = day;
        this.monthTriggering = month;
        this.yearTriggering = year;
    }

    public void resetSettings() {
        this.dayTriggering = DEFAULT_DAY_TRIGGERING;
        this.monthTriggering = DEFAULT_MONTH_TRIGGERING;
        this.yearTriggering = DEFAULT_YEAR_TRIGGERING;
        this.mode = PASSIVE_MODE;
    }
}